starlite/contrib/htmx/request.py

```diff
@@ -0,0 +1,106 @@
+        """Initialize Class"""
+        self.request = request
+
+    def _get_header_value(self, name: str) -> Optional[str]:
+        """Parse request header
+        Checks for uri encoded header and unquotes it in readable format.
+        """
+        value = self.request.headers.get(name) or None
+        if value and self.request.headers.get(f"{name}-uri-autoencoded") == "true":
+            return unquote(value)
+        return value
+
+    def __bool__(self) -> bool:
+        """Allow to check whether request is sent by a HTMX client."""
+        return self._get_header_value("hx-request") == "true"
+
+    @cached_property
+    def boosted(self) -> bool:
+        """Allow to check whether request is boosted."""
+        return self._get_header_value("hx-boosted") == "true"
+
+    @cached_property
+    def current_url(self) -> Optional[str]:
+        """Current url value sent by HTMX client. Helps in tracking navigation history."""
+        return self._get_header_value("hx-current-url")
+
+    @cached_property
+    def current_url_abs_path(self) -> Optional[str]:
+        """Current url abs path value, to get query and path parameter sent by HTMX client."""
+        url = self.current_url
+        if url is not None:
+            split = urlsplit(url)
+            if split.scheme == self.request.scope["scheme"] and split.netloc == self.request.headers.get("host"):
+                return urlunsplit(split._replace(scheme="", netloc=""))
+            return None
+        return url
+
+    @cached_property
+    def history_restore_request(self) -> bool:
+        """If True then, request is for history restoration after a miss in the local history cache"""
+        return self._get_header_value("hx-history-restore-request") == "true"
+
+    @cached_property
+    def prompt(self) -> Optional[str]:
+        """User Response to prompt.
+        <button hx-delete="/account" hx-prompt="Enter your account name to confirm deletion">
+            Delete My Account
+        </button>
+        """
+        return self._get_header_value("hx-prompt")
+
+    @cached_property
+    def target(self) -> Optional[str]:
+        """The id of the target element if it exists"""
+        return self._get_header_value("hx-target")
+
+    @cached_property
+    def trigger(self) -> Optional[str]:
+        """The id of the triggered element if it exists"""
+        return self._get_header_value("hx-trigger")
+
+    @cached_property
+    def trigger_name(self) -> Optional[str]:
+        """The name of the triggered element if it exists"""
+        return self._get_header_value("hx-trigger-name")
+
+    @cached_property
+    def triggering_event(self) -> Any:
+        """The name of the triggered event.
+        'event-header' extension adds the Triggering-Event header to requests.
+        """
+        value = self._get_header_value("triggering-event")
+        if value is not None:
+            try:
+                value = json.loads(value)

```

Comment:
replace this logic with the `decode_json` function exposed by Starlite (`starlite.utils.serialization.decode_json`)